dist/dagre.js

```diff
@@ -467,6 +467,127 @@ var _ = require("./lodash"),
 
 module.exports = layout;
 
+function hasMultiplePorts(g, node) {
+  var inPorts = {};
+  var outPorts = {};
+
+  _.forEach(g.nodeEdges(node), function (edge) {
+    if (edge.v === node && edge.vport) {
+      inPorts[edge.vport] = {};
+    } else if (edge.w === node && edge.wport) {
+      outPorts[edge.wport] = {};
+    }
+  });
+
+  return  _.size(inPorts) > 1 || _.size(outPorts) > 1;
+}
+
+function addDummyPortNode(g, parent, port, order) {
+  var dummyNode = util.addDummyNode(g, "port", {}, "port");
+  var node = g.node(dummyNode);
+  node.port = port;
+  node.rank = g.node(parent).rank;
+  g.setParent(dummyNode, parent);
+  return dummyNode;
+}
+
+function addPortNodes(g, node) {
+  // Create the port nodes
+  var portNodes = {};
+  _.forEach(g.node(node).ports, function (port) {
+    portNodes[port] = addDummyPortNode(g, node, port);
+  });
+  // Set the order of children to be fixed
+  g.node(node).fixedOrder = true;
+
+  // Hook up the edges to the port nodes
+  _.forEach(g.nodeEdges(node), function (edge) {
+    var port;
+    if (edge.v === node && edge.vport) {
+      var dummyNode = portNodes[edge.vport];
+      g.setEdge(dummyNode, edge.w, {
+        weight: 1,
+        dummy: "port",
+        edge: edge,
+        edgeValue: g.edge(edge)
+      });
+      g.removeEdge(edge);
+      // TODO: Handle children.
```

Comment:
Could you put your name after the TODO, it makes it easier in the future to know who put it there. 
```suggestion
      // TODO DeanLanier: Handle children.
```
Also, the insert a suggestion is new, so I wanted to try it

Reply:
I think I will just remove the TODO comment and add a note explaining that nodes with children aren't supported with these changes.